src/doc/documentation-guide.md

```diff
@@ -94,6 +95,68 @@ that concatenates the various `README.md` files we have.
 
 # Stylistic Conventions
 
+## General Considerations
+
+Documentation is most effective when it is precise, concise and uniform.
+
+ - **Preciseness**. Vague descriptions do more harm than good as different
```

Comment:
How about a link to some widely accepted technical writing style guide or a list of pithy quotes? I really do think that "preachy" is a tone that we absolutely want to avoid.

I know that I am a broken record but I still feel like the core problem we need to fix re: documentation is to make it more usable/accessible. Otherwise it just feels like a tax.